elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLColumnProjection.java

```diff
@@ -41,6 +41,14 @@ default String toSQL(Queryable source, SQLReferenceTable lookupTable) {
     default boolean canNest(Queryable source, SQLReferenceTable lookupTable) {
         SQLDialect dialect = source.getConnectionDetails().getDialect();
         String sql = toSQL(source.getSource(), lookupTable);
+
+        // TODO: Remove this when switching to TableContext
+        // Resolved Reference would be empty if value is not provided in @MetricFormula as value is optional.
+        // Once we change this to use TableContext, then it should be able to resolve expression directly.
+        if (this instanceof SQLMetricProjection && sql.equals("")) {
```

Comment:
Instead of this hack here, why not just change DailyAverageScorePerPeriod to override canNest to true.